apps/api/src/modules/job/job.service.ts

```diff
@@ -20,34 +23,117 @@ export class JobService {
-        // maxWage: searchJobDto.maxWage || undefined,
-
-        //TODO: will find out how to filter enum later [24]
-        //status: searchJobDto.status || undefined,
-        //gender: searchJobDto.gender || undefined,
-
+        //job always have shooting
+        Shooting: {
+          every: {
+            startDate: {
+              gte: searchJobDto.startDate || defaultStartDate,
+            },
+            endDate: {
+              lte: searchJobDto.endDate || defaultEndDate,
+            },
+            startTime: {
+              gte: searchJobDto.startTime || defaultStartTime,
+            },
+            endTime: {
+              lte: searchJobDto.endTime || defaultEndTime,
+            },
+          },
+          some: {
+            shootingLocation: searchJobDto.location || undefined,
+          },
+        },
+
+        minAge: {
+          lte: Number(searchJobDto.age) || defaultminAgeLte,
+        },
+        maxAge: {
+          gte: Number(searchJobDto.age) || defaultMaxAgeGte,
+        },
+
+        wage: {
+          gte: Number(searchJobDto.minWage) || defaultMinWageGte,
+          lte: Number(searchJobDto.maxWage) || defaultMaxWageLte,
+        },
+
+        status: undefined,
+        gender: undefined,
         castingId: Number(searchJobDto.castingId) || undefined,
       },
     }
+    //handle array and undefined
+    searchJobDto.status = searchJobDto.status || [SearchJobStatus.OPEN]
+    //check searchJobDto.status is array or not
+    if (!Array.isArray(searchJobDto.status)) {
+      searchJobDto.status = [searchJobDto.status]
+    }
+
+    //handle status qyery
+    const statusQuery = []
+    //make OPEN for default
+    searchJobDto.status = searchJobDto.status || [SearchJobStatus.OPEN]
+    if (searchJobDto.status.includes(SearchJobStatus.OPEN)) {
+      statusQuery.push(JobStatus.OPEN)
+    }
+    if (searchJobDto.status.includes(SearchJobStatus.CLOSE)) {
+      statusQuery.push(
+        JobStatus.SELECTING,
+        JobStatus.SELECTION_ENDED,
+        JobStatus.FINISHED,
+      )
+      if (user.type != UserType.ACTOR) {
+        statusQuery.push(JobStatus.CANCELLED)
+      }
+    }
+    params.where.status = {
+      in: statusQuery,
+    }
+
+    //handle gender qyery
+    if (searchJobDto.gender) {
+      //check searchJobDto.gender is array or not (the case that only one params it will not be array)
+      if (!Array.isArray(searchJobDto.gender)) {
+        searchJobDto.gender = [searchJobDto.gender]
+      }
+      //if Query only Gender.ANY return all
+      if (
+        !(
+          searchJobDto.gender &&
+          searchJobDto.gender.length === 1 &&
+          searchJobDto.gender[0] === Gender.ANY
+        )
+      ) {
+        params.where.gender = {
+          in: [...searchJobDto.gender, Gender.ANY],
+        }
+      }
+    }
+    //handle location undefined
+    if (searchJobDto.location) {
+      params.where.Shooting.some = {
+        shootingLocation: searchJobDto.location,
```

Comment:
Should this be substring search? @miello

Reply:
I think it might be convenient if we use substring search